libs/shared/sai-editor/src/sai-editor.component.ts

```diff
@@ -91,6 +93,13 @@ export class SaiEditorComponent extends MultiRowEditorComponent<SmartScripts> im
   public override readonly editorService = inject(SaiEditorService);
   protected override readonly handlerService = inject(SaiHandlerService);
 
+  selectedName$!: Observable<string>;
+
+  override ngOnInit(): void {
+    super.ngOnInit();
+    this.selectedName$ = this.handlerService.getName().pipe(shareReplay(1));
```

Comment:
I like the idea of using `shareReplay(1)`, however shall we try using it directly inside the sai-handler service?

something like:

```
    return this.queryService.query<{ name: string }>(query).pipe(
      map((data) => {
        if (data.length > 0) {
          return `${data[0].name}`;
        } else {
          console.error(`Unable to find name for source_type = ${sai.source_type}, entryorguid = ${sai.entryorguid}`);
          return '';
        }
      }),
      shareReplay(1), // <--- ADD THIS HERE
    );
```

then there is no need for changes in the component or in other consumers of that service